src/algebra/group_action_hom.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import algebra.group_ring_action
+
+/-!
+# Equivariant homomorphisms
+
+## Main definitions
+
+* `mul_action_hom M X Y`, the type of equivariant functions from `X` to `Y`, notated `X →[M] Y`.
+* `distrib_mul_action_hom M A B`, the type of equivariant additive monoid homomorphisms
+  from `A` to `B`, notated `A →+[M] B`.
+* `mul_semiring_action_hom M R S`, the type of equivariant ring homomorphisms
+  from `R` to `S`, notated `R →+*[M] S`.
+-/
+
+universes u v w u₁
+
+variables (M : Type u) [monoid M]
+variables (X : Type v) [mul_action M X]
+variables (Y : Type w) [mul_action M Y]
+variables (Z : Type u₁) [mul_action M Z]
+variables (A : Type v) [add_monoid A] [distrib_mul_action M A]
+variables (A' : Type v) [add_group A'] [distrib_mul_action M A']
+variables (B : Type w) [add_monoid B] [distrib_mul_action M B]
+variables (B' : Type w) [add_group B'] [distrib_mul_action M B']
+variables (C : Type u₁) [add_monoid C] [distrib_mul_action M C]
+variables (R : Type v) [semiring R] [mul_semiring_action M R]
+variables (R' : Type v) [ring R'] [mul_semiring_action M R']
+variables (S : Type w) [semiring S] [mul_semiring_action M S]
+variables (S' : Type w) [ring S'] [mul_semiring_action M S']
+variables (T : Type u₁) [semiring T] [mul_semiring_action M T]
+variables (G : Type u) [group G] (H : set G) [is_subgroup H]
+
+set_option old_structure_cmd true
+
+/-- Equivariant functions. -/
+@[nolint has_inhabited_instance]
+structure mul_action_hom :=
+(to_fun : X → Y)
+(map_smul' : ∀ (m : M) (x : X), to_fun (m • x) = m • to_fun x)
+
+notation β ` →[`:25 α:25 `] `:0 γ:0 := mul_action_hom α β γ
```

Comment:
Is there a reason you are using greek letters here?

Reply:
I just copied it from `linear_map`

Reply:
Meh... I'd rather see the greek letter be removed over there as well (other PR).

Reply:
done